### ISSUE TITLE
Include template does not support class inheritance

### ISSUE THREAD

**Issue by sam0delkin** [nelmio/alice#276]
I have some base class (User), and a child class (AdminUser), that extends base class.
When i created a template for base class:
```yaml
Acme\UserBundle\Entity\User:
    user (template):
        firstName: <firstName()>
        lastName: <lastName()>
```
and then include it in another file:
```yaml
include:
    - user.yml

Acme\AdminBundle\Entity\AdminUser:
    adminUser (extends user):
        email (unique): admin@example.com
        plainPassword: passw0rd
        enabled: true
```
and then trying to load **only second file**, i have this error: `Template user is not defined.`
But if i load a template file first, then no error shown.
After some investigation, i found out that this is related to this string of documentation (https://github.com/nelmio/alice/blob/master/doc/fixtures-refactoring.md#including-files):
`All files are merged in one data set before generation, and the includer's content takes precedence over included files' fixtures in case of duplicate keys.`

So, maybe some additional code should be implemented to support class inheritance in template loading?

**Comment by theofidry** [comment]
@sam0delkin did you try to have your parser load the two files at the same time rather than having an import statement and loading one file?

**Comment by sam0delkin** [comment]
@theofidry maybe my english is not so easy to understand, sorry about that  :), but:
> But if i load a template file first, then no error shown.

so, yes, if i load both files, then everything is working great. But then why i need to use the includes in template? I just can load both files and working with parent template, correct?

**Comment by sam0delkin** [comment]
@theofidry let me explain it more correctly. For example, we have 3 files:
* user.yml:
```yaml
Acme\UserBundle\Entity\User:
    user (template):
        firstName: <firstName()>
        lastName: <lastName()>
```
* admin_user.yml:
```yaml
include:
    - user.yml

Acme\AdminBundle\Entity\AdminUser:
    adminUser (extends user):
        email (unique): admin@example.com
        plainPassword: passw0rd
        enabled: true
```
* test.php
```php
include '../vendor/autoload.php';

$loader = new Nelmio\Alice\Loader\Yaml();
$loader->load('admin_user.yml');
```
And let's see the **parsed** (not loaded) data in this case:
```
array(2) {
  'Acme\AdminBundle\Entity\AdminUser' =>
  array(1) {
    'adminUser (extends user)' =>
    array(3) {
      'email (unique)' =>
      string(17) "admin@example.com"
      'plainPassword' =>
      string(8) "passw0rd"
      'enabled' =>
      bool(true)
    }
  }
  'Acme\UserBundle\Entity\User' =>
  array(1) {
    'user (template)' =>
    array(2) {
      'firstName' =>
      string(13) "<firstName()>"
      'lastName' =>
      string(12) "<lastName()>"
    }
  }
}
```
in this case we will have an error: `Fatal error: Uncaught exception 'UnexpectedValueException' with message 'Template user is not defined.'`
**BUT**
if we will change the template file (user.yml) to use **child** and not **base** class:
```yaml
Acme\AdminBundle\Entity\AdminUser:
    user (template):
        firstName: <firstName()>
        lastName: <lastName()>
```
then we will have this **parsed** data:
```
array(1) {
  'Acme\AdminBundle\Entity\AdminUser' =>
  array(2) {
    'user (template)' =>
    array(2) {
      'firstName' =>
      string(13) "<firstName()>"
      'lastName' =>
      string(12) "<lastName()>"
    }
    'adminUser (extends user)' =>
    array(3) {
      'email (unique)' =>
      string(17) "admin@example.com"
      'plainPassword' =>
      string(8) "passw0rd"
      'enabled' =>
      bool(true)
    }
  }
}
```
so, in case of using class inheritance templates including is working not correct.

**Comment by theofidry** [comment]
```php
Fixtures::load(['user.yml', 'user_admin.yml']; $objectManager);
```

without the import statement?

**Comment by oddz** [comment]
The documentation should at least note that includes aren't compatible with inheritance.

**Comment by oddz** [comment]
I have confirmed that you can't include templates and expect them to work for inheritance. The reason is that the includes are appended to the end of the data array specified in the below method. Therefore, the data within the includes are not available in the builder until all the standard fixtures have been processed.

Nelmio\Alice\Fixtures\Parser\Methods\Yaml::parse()

**Comment by oddz** [comment]
This should definitely be noted in the docs because it seems to me like including templates that are used for inheritance are a very likely scenario. One that is not currently supported.

**Comment by theofidry** [comment]
@sam0delkin could you please check if this has been solved by #359?

**Comment by theofidry** [comment]
The issue has been solved in 2.2.0 :)
